app/static/js/app.js

```diff
@@ -223,6 +227,9 @@ document.addEventListener('DOMContentLoaded', function(){
                                 self.translatedText = res.translatedText;
                                 self.loadingTranslation = false;
                                 self.output = JSON.stringify(res, null, 4);
+                                if(self.sourceLang == "auto" && res.detectedLanguage.length > 0){
```

Comment:
there is a problem here, detectedLanguage is an object we cannot use .length because it returns undefined on an object
so the condition never triggers

Reply:
```js
if(self.sourceLang == "auto" && res.detectedLanguage !== undefined){
```

this solve the problem

Reply:
Ah, good catch @dingedi . I should have reviewed more thoroughly. Can you make a commit into main?

Reply:
yes done in https://github.com/LibreTranslate/LibreTranslate/commit/5d8e513d45e4764189b3b7516e5b9f29e6a6f38e

Reply:
Thanks for noticing & fixing @dingedi!